packages/rich-text/src/to-tree.js

```diff
@@ -33,27 +10,73 @@ export function toTree( value, multilineTag, settings ) {
 		appendText,
 		onStartIndex,
 		onEndIndex,
-		onEmpty,
 	} = settings;
 	const { formats, text, start, end } = value;
 	const formatsLength = formats.length + 1;
-	const tree = createEmpty( tag );
+	const tree = createEmpty();
+	const multilineFormat = { type: multilineTag };
 
-	append( tree, '' );
+	let lastCharacterFormats;
+	let lastCharacter;
+
+	// If we're building a multiline tree, start off with a multiline element.
```

Comment:
Why?

This almost reads in direct contrast to what was discussed at https://github.com/WordPress/gutenberg/pull/10678#discussion_r226456573 .

Reply:
No, this is purely for the creation of an *editable* DOM tree, which needs the multiline element if there is one only. If we change the output HTML, it will be changed in `toHTMLString`, which is purely for the output.

Reply:
The "Why" should probably be made clear in the code comment. The "What" is usually pretty obvious already anyways, so doesn't add much on its own as a comment.